src/Symfony/Component/Config/Definition/BaseNode.php

```diff
@@ -198,12 +198,41 @@ public function setRequired(bool $boolean)
     /**
      * Sets this node as deprecated.
      *
+     * @param string $package The name of the composer package that is triggering the deprecation
+     * @param string $version The version of the package that introduced the deprecation
+     * @param string $message The deprecation message to use
+     *
      * You can use %node% and %path% placeholders in your message to display,
      * respectively, the node name and its complete path.
      */
-    public function setDeprecated(?string $message)
+    public function setDeprecated(?string $package/*, string $version, string $message = 'The child node "%node%" at path "%path%" is deprecated.' */)
```

Comment:
Adding the parameters at the front makes it impossible to both support 5.1+ and <5.1 as a bundle.

If you use the intended 5.1+ way:
`->setDeprecated("my/bundle", "1.0", "this node is deprecated, do sth else")`

* in `5.1+` we will show the message `"this node is deprecated, do sth else"`
* in `<5.1` we will show the message `"my/bundle"`

Is there a (non-hacky) way to have the real message in both versions?

@nicolas-grekas @atailouloute

Reply:
I just created https://github.com/symfony/symfony/issues/37284 to track this